src/services/ui/src/pages/actions/setups/setupWithdrawPackage.ts

```diff
@@ -6,12 +6,28 @@ import {
 } from "@/pages/form/zod";
 
 export const medicaidWithdrawPackageSetup = {
-  schema: z.object({
-    additionalInformation: zAdditionalInfo,
-    attachments: z.object({
-      supportingDocumentation: zAttachmentOptional,
+  schema: z
+    .object({
+      additionalInformation: zAdditionalInfo.optional(),
+      attachments: z.object({
+        supportingDocumentation: zAttachmentOptional,
+      }),
+    })
+    .superRefine((val, ctx) => {
+      if (
+        !val.attachments.supportingDocumentation?.length &&
+        val.additionalInformation === undefined
+      ) {
+        ctx.addIssue({
+          message: "An Attachment or Additional Information is required.",
+          code: z.ZodIssueCode.custom,
+          fatal: true,
+        });
+        // Zod says this is to appease types
+        // https://github.com/colinhacks/zod?tab=readme-ov-file#type-refinements
+        return z.NEVER;
+      }
```

Comment:
This is truly an amazing find. I have nothing bad to say. The ability to keep our validations all in one place will, one speed up the app, but also make it much simpler